ChronoVisor/src/chronovisor_instance.cpp

```diff
@@ -36,39 +33,41 @@ int main(int argc, char**argv)
     // If we can't ensure the instantiation of registries on the main thread
     // we'd need to add static mutex protection to all the registry singleton objects
 
+    Logger::initialize("file", "../../logs/ChronoVisor_initlogfile.txt", spdlog::level::debug, "ChronoVisor");
+    Logger::getLogger()->info("[chronovisor_instance] Init configuration process.");
     std::string default_conf_file_path = "./default_conf.json";
     std::string conf_file_path;
     conf_file_path = parse_conf_path_arg(argc, argv);
     if(conf_file_path.empty())
     {
         conf_file_path = default_conf_file_path;
     }
-
     ChronoLog::ConfigurationManager confManager(conf_file_path);
+    Logger::getLogger()->info("[chronovisor_instance] Configuration process completed.");
 
-    chronolog::VisorClientPortal theChronoVisorPortal; // confManager);
+    Logger::changeConfiguration(confManager.VISOR_CONF.VISOR_LOG_CONF.LOGTYPE
+                                , confManager.VISOR_CONF.VISOR_LOG_CONF.LOGFILE
+                                , confManager.VISOR_CONF.VISOR_LOG_CONF.LOGLEVEL
+                                , confManager.VISOR_CONF.VISOR_LOG_CONF.LOGNAME);
+    Logger::getLogger()->info("[chronovisor_instance] Running Chronovisor Server.");
 
+    chronolog::VisorClientPortal theChronoVisorPortal;
     chronolog::KeeperRegistry keeperRegistry;
 
     // ChronoVisor::ChronoVisorServer2 visor(confManager);
-
     keeperRegistry.InitializeRegistryService(confManager);//provider_id=22);
-
-//    visor.start(&keeperRegistry);
-
+    // visor.start(&keeperRegistry);
     theChronoVisorPortal.StartServices(confManager, &keeperRegistry);
 
-
     /////
-
+    Logger::getLogger()->info("[chronovisor_instance] ChronoVisor Running...");
     while(keep_running)
     {
         sleep(10);
     }
 
-
     theChronoVisorPortal.ShutdownServices();
     keeperRegistry.ShutdownRegistryService();
-
+    Logger::getLogger()->info("[chronovisor_instance] ChronoVisor shutdown.");
```

Comment:
logger object needs to be destroyed, as it's created on the heap

Reply:
I have chosen to keep the destructor private because the class is managed by a std::shared_ptr. This approach automatically handles the logger's lifecycle, including its destruction when the application terminates or when all references are out of scope. This design aligns with the singleton pattern, ensuring controlled and safe management of the logger's instance without requiring public access to the destructor. In any case, do you prefer to make it public?